lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -825,4 +853,163 @@ class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
       }
     }
   }
+
+  Recipe? _recipeOnDevice(recipeNumber) {
+    List<Recipe> deviceRecipes = List<Recipe>.from(state.recipes);
+    for (Recipe recipe in deviceRecipes) {
+      if (recipe.recipeNumber == recipeNumber) {
+        return recipe;
+      }
+    }
+    return null;
+  }
+
+  Future<void> _onRecipeImported(
+      RecipeImported event, Emitter<RecipeState> emit) async {
+    emit(state.copyWith(recipeImportStatus: '')); // Clear status for snackbar
```

Comment:
Is this needed? Wouldn't it be enough to just change the status at the end? (Haven't tried that just asking 😄)

Reply:
There are situations where the previous result is equal to the new result - e.g we previously imported one recipe and then proceeded to import one recipe again. Since the state is the same the snackbar would not show when we would still want to display to the user that the recipe was indeed imported - resetting the state solves this issue.

Reply:
I see, okay!